src/page-modules/assistant/trip/index.tsx

```diff
@@ -3,48 +3,46 @@ import {
   formatToWeekday,
   parseIfNeeded,
 } from '@atb/utils/date';
-import {
-  TripData,
-  type TripPattern as TripPatternType,
-} from '../server/journey-planner/validators';
 import style from './trip.module.css';
 import { PageText, useTranslation } from '@atb/translations';
 import { Typo } from '@atb/components/typography';
-import { GeocoderFeature } from '@atb/page-modules/departures';
-import { nextTripPatterns } from '../client';
-import { FromToTripQuery, NonTransitTripData } from '../types';
-import {
-  createTripQuery,
-  filterOutDuplicates,
-  getCursorBySearchMode,
-} from '../utils';
+import { useNonTransitTrip, useTripPatterns } from '../client';
+import { FromToTripQuery } from '../types';
+import { createTripQuery } from '../utils';
 import { useEffect, useState } from 'react';
 import { Button } from '@atb/components/button';
 import { NonTransitTrip } from '../non-transit-pill';
 import { isSameDay } from 'date-fns';
 import { capitalize } from 'lodash';
 import EmptySearchResults from '@atb/components/empty-message';
 import TripPattern from './trip-pattern';
-import { SearchTime } from '@atb/modules/search-time';
-import {
-  getInitialTransportModeFilter,
-  type TransportModeFilterOption,
-} from '@atb/modules/transport-mode';
+import { getInitialTransportModeFilter } from '@atb/modules/transport-mode';
+import EmptySearch from '@atb/components/loading-empty-results';
 
 export type TripProps = {
   tripQuery: FromToTripQuery;
-  trip: TripData;
-  nonTransitTrips: NonTransitTripData;
 };
 
-export default function Trip({ tripQuery, trip, nonTransitTrips }: TripProps) {
+export default function Trip({ tripQuery }: TripProps) {
+  const [query, setQuery] = useState(
+    createTripQuery(tripQuery, getInitialTransportModeFilter(null)),
+  );
   const { t } = useTranslation();
-  const { tripPatterns, fetchNextTripPatterns, isFetchingTripPatterns } =
-    useTripPatterns(trip, tripQuery.searchTime);
+  const { trips, isLoading, loadMore, isLoadingMore } = useTripPatterns(query);
+  const { nonTransitTrips } = useNonTransitTrip(query);
 
-  const nonTransits = Object.entries(nonTransitTrips);
+  useEffect(() => {
+    const q = createTripQuery(tripQuery, getInitialTransportModeFilter(null));
+    setQuery(q);
+  }, [tripQuery]);
```

Comment:
Is the intention here to avoid query changing as object causing an infinite fetch loop? If so I think maybe `useMemo` is a better option? 🤔

Reply:
This might be a remnant from several rounds of refactoring when transitioning from fetch to useSWR. That was the intention, and `useMemo` should be a better option to solve that. However, it might not be necessary to parse the `FromToTripQuery` object here at all 🤔 I guess the hook can handle that as well. This also makes the transport mode filter not work, so I'll fix that 👍